File: JavaScript/23_Math/LC_1134_ArmstrongNumber.js
/**
 * @param {number} n
 * @return {boolean}
 */
var isArmstrong = function(n) {
    var ori = n;
    var bit = String(n).split('')
    var sum = 0;
    var len = bit.length;
    for(var i = 0; i < len; i++){
        sum = sum + Math.pow(Number(bit[i]),len);
    }
    return sum === ori;
};

File: Java/23_Math/LC_1134_ArmstrongNumber.java
class Solution_1134 {
    public boolean isArmstrong(int n) {
        int num = n;
        String s_ori = String.valueOf(num);
        int len = s_ori.length();
        int sum = 0;
        for(int i = 0; i < len; i++){
            sum += Math.pow(num % 10,len);
            num = num / 10; 
        }

        return sum == n;
    }
}

public class LC_1134_ArmstrongNumber {
    
}